ciphersuite/ed25519.go

```diff
@@ -0,0 +1,221 @@
+package ciphersuite
+
+import (
+	"encoding/hex"
+	"io"
+
+	"golang.org/x/crypto/ed25519"
+	"golang.org/x/xerrors"
+)
+
+const errNotEd25519CipherSuite = "invalid cipher suite"
+const errInvalidBufferSize = "invalid buffer size"
+
+// Ed25519CipherSuiteName is the name of the cipher suite that is using Ed25519 as
+// the signature algorithm.
+const Ed25519CipherSuiteName = "ED25519_CIPHER_SUITE"
+
+// Ed25519PublicKey is the public key implementation for the Ed25519 cipher suite.
+type Ed25519PublicKey struct {
```

Comment:
Why using a struct and not the type directly?

Reply:
You mean the ed25519 type ? Because I need to implement our abstraction that is used in Onet.

Reply:
Ho, okay, I'm discovering a bit more of golang, thanks